crates/sled/src/frag.rs

```diff
@@ -17,3 +20,16 @@ pub(crate) enum Frag {
     ParentMergeConfirm,
     ChildMergeCap,
 }
+
+impl Frag {
+    pub fn base(data: Data) -> Frag {
```

Comment:
could you rename this function `root` to reduce the chances of misuse over time? I want to be clear that it's not really valid for other purposes

Reply:
@spacejam did you mean rename this function to `root`?

Reply:
yes, please. the reason is that the only kind of base that this creates a valid version of is for new tree root nodes. you could also assert that data.is_index() to give extra certainty

Reply:
there is also a Leaf at crates/sled/src/meta.rs:35

Reply:
@spacejam I have made a `base` private and exposed two methods `leaf` and `root` with assertions. Hope I get your point well. If not, don't hesitate to ping me back.